src/main/java/net/minecraftforge/common/capabilities/CapabilityManager.java

```diff
@@ -55,30 +54,56 @@
      * @param type The class type to be registered
      * @deprecated use {@link RegisterCapabilitiesEvent}
      */
-    @Deprecated(forRemoval = true)
+    @Deprecated(since = "1.18", forRemoval = true)
     public <T> void register(Class<T> type)
     {
         Objects.requireNonNull(type,"Attempted to register a capability with invalid type");
-        String realName = type.getName().intern();
+        get(Type.getType(type).toString(), true);
```

Comment:
I don't think this line is correct. `Type#toString` says it returns the descriptor, which would be `Lfoo/bar/Baz;`.
It should be
```suggestion
        get(type.getName(), true);
```
like in `RegisterCapabilitiesEvent`